spec/request/mhv_accounts_request_spec.rb

```diff
@@ -29,135 +29,261 @@
-          get v0_mhv_account_path
-          expect(response).to be_success
-          expect(JSON.parse(response.body)['data']['attributes']['account_state']).to eq('existing')
+      context 'that is' do
+        %w[Basic Advanced].each do |type|
+          context type do
+            around(:each) do |example|
+              VCR.use_cassette('mhv_account_type_service/premium') do
+                VCR.use_cassette("mhv_account_type_service/#{type.downcase}") do
+                  example.run
+                end
+              end
+            end
+
+            it_behaves_like 'a successful GET #show', account_state: 'existing', account_level: type
+            it_behaves_like 'ssn mismatch', account_level: type
+            it_behaves_like 'non va patient', account_level: type
+            it_behaves_like 'a successful POST #upgrade'
+            it_behaves_like 'a failed POST #create', http_status: :forbidden,
+                                                     message: V0::MhvAccountsController::CREATE_ERROR
+          end
         end
 
-        it 'raises error for POST #create' do
-          post v0_mhv_account_path
-          expect(response).to have_http_status(:forbidden)
+        %w[Premium Error Unknown].each do |type|
+          context type do
+            around(:each) do |example|
+              VCR.use_cassette("mhv_account_type_service/#{type.downcase}") do
+                example.run
+              end
+            end
+
+            it_behaves_like 'a successful GET #show', account_state: 'existing', account_level: type
+            it_behaves_like 'ssn mismatch', account_level: type
+            it_behaves_like 'non va patient', account_level: type
+            it_behaves_like 'a failed POST #create', http_status: :forbidden,
+                                                     message: V0::MhvAccountsController::CREATE_ERROR
+            it_behaves_like 'a failed POST #upgrade', http_status: :forbidden,
+                                                      message: V0::MhvAccountsController::UPGRADE_ERROR
+          end
         end
       end
 
       context 'that is registered' do
         before(:each) do
-          mhv_account = MhvAccount.find_or_initialize_by(user_uuid: user.uuid)
-          mhv_account.update(account_state: 'registered', registered_at: Time.current)
+          MhvAccount.skip_callback(:initialize, :after, :setup)
+          MhvAccount.create(user_uuid: user.uuid, mhv_correlation_id: mhv_ids.first,
+                            account_state: 'registered', registered_at: Time.current)
+          MhvAccount.set_callback(:initialize, :after, :setup)
         end
 
-        it 'responds to GET #show' do
-          get v0_mhv_account_path
-          expect(response).to be_success
-          expect(JSON.parse(response.body)['data']['attributes']['account_state']).to eq('registered')
+        around(:each) do |example|
+          VCR.use_cassette('mhv_account_type_service/premium') do
```

Comment:
Also here, wondering why the Premium account type VCR is wrapped around the Advanced.

Reply:
i'm not sure from the snippet if this is a SHOW action or not, but the idea is that if the account is upgraded the serialized JSON should return the account level (the correct one). Is this necessary to test, probably not.